### DISCUSSION TITLE
How can I enable Scrubber to also take the dictionary keys into account?

### DISCUSSION THREAD

**Discussion by samtrion** [VerifyTests/Verify#1289] [General]
I am currently trying to expand my tests further. For this I need a scrubber, which removes inline `Version`, as they occur for example in `typeof(string).FullName`. Unfortunately, I have only moderate success with this, as not all content is corrected.

For that purpose i wrote the following code...

```csharp
private static void ReplaceVersions(StringBuilder builder)
{
    const string version = "Version=";

    if (builder.Length <= version.Length)
    {
        return;
    }

    var indexStart = 0;
    do
    {
        var value = builder.ToString();
        var indexVersion = value.IndexOf(
            version,
            indexStart,
            StringComparison.OrdinalIgnoreCase
        );
        if (indexVersion == -1)
        {
            return;
        }

        var indexEnd = value.IndexOf(',', indexVersion);
        if (indexEnd == -1)
        {
            return;
        }
        _ = builder
            .Remove(indexVersion, indexEnd - indexVersion)
            .Insert(indexVersion, $"{version}x.x.x.x");
        indexStart = indexVersion + version.Length;
    } while (indexStart < builder.Length);
}
```
However, this only works with individual properties, not with dictionary keys.

![image](https://github.com/user-attachments/assets/6d5b7cd5-b42f-4acb-a7ba-04b0be441805)

Any solution?

**Comment by SimonCropp** [comment]
what type is the key?

**Reply by samtrion** [reply]
The key is `string`

I mainly use `System.Collections.Generic.Dictionary<string, CustomRecord>` in the context of .NET versions 6 and 7 and starting with .NET8.0 `System.Collections.Frozen.FrozenDictionary<string, CustomRecord>`.

**Reply by samtrion** [reply]
@SimonCropp Any ideas?

**Reply by SimonCropp** [reply]
yeah it is a bug. i am working on it. eta a few days

**Reply by SimonCropp** [reply]
can u try the new nuget

**Reply by samtrion** [reply]
Works, as intended.

https://github.com/dailydevops/archiduct/pull/59
